src/core_scan/Core_scan.ml

```diff
@@ -921,8 +954,9 @@ let select_applicable_rules_for_target ~analyzer ~products ~path
             * strategy.
          *)
          match r.R.paths with
-         | Some paths when respect_rule_paths ->
-             Filter_target.filter_paths paths path
+         | Some paths when respect_rule_paths -> (
```

Comment:
Should we be doing this based on the source? What if the source is some relative path that doesn't get correctly filtered when an absolute path would?

Reply:
Yes, this should be based on the source so that when we have non-file sources we can correctly apply include/exclude to them rather than the tempfile path. We already implicitly add a `**/` and `/**` before and after (resp.) the pattern, and this is documented to be relative to the project root (https://semgrep.dev/docs/writing-rules/rule-syntax/#paths), so I believe this is sound.

As a motivating example for this, consider a git-based source:
```ocaml
| GitBlob { paths = target_paths; _ } ->
    List.exists
      (fun (_commit_sha, path_at_commit) ->
        Filter_target.filter_paths paths path_at_commit)
      target_paths)
```
where we want to run if the blob is present at some commit with a path indicating it should be run against. If we ran against the file field then we wouldn't be able to have this behaviour.

Reply:
Added a comment in 72603150d9f141f50d0ac51a07fef6c9bd2bc2ce